website/content/en/guides/level-up/csv-enrichment-guide.md

```diff
@@ -0,0 +1,210 @@
+Before you begin, this guide assumes the following:
+
+* You understand the [basic Vector concepts][concepts]
+* You understand [how to set up a basic pipeline][pipeline]
+
+[concepts]: /docs/about/concepts
+[pipeline]: /docs/setup/quickstart
+{{< /requirement >}}
+
+
+It's important for any organization maintaining observability pipelines to
+have flexibility to manipulate their observability data — whether it be appending
+better context to an event or triggering alerts when there is a potential threat.
+A key component that drives that flexibility is enriching your data from different
+sources. Vector now offers initial support for enriching events from external data
+sources, which is currently powered by a powerful Vector concept,
+[enrichment tables][Enrichment tables]. For now, our support for enrichment
+through external data sources is limited to `csv` files, but we're looking to
+add support for more data sources.
+
+This guide walks through how you can start enriching your observability by
+ exploring two interesting use cases.
+
+## Use Case 1 - IoT Devices
+
+When working with IoT devices, you often want to minimize payloads being
+emmitted by the devices, even when collecting events from those devices. This
+is where [Enrichment tables] comes in very handy. Let's assume that you have an
+IoT device that emits its status - `online`, `offline`, `transmitting`, and
+`error` states, but the device needs to emit as little payload as possible
+due to technical constraints or requirements; instead, the IoT can device can
+emit integers — `1` for `online`, `2` for `offline`, etc.
+
+To enrich the IoT device's observability data, let's use a `csv` file
+containing the following information:
+
+```csv
+status_code,status_message
+1,"device status online"
+2,"device status offline"
+3,"device status connection error"
+...
+7,"device status transmitting"
+8,"device status transmission complete"
+9,"device status not responding"
+```
+
+To enrich your observability data, you can use ['get_enrichment_table_record'][get_enrichment_table_record].
+Assuming that your `csv` file is called `iot_status.csv`, the following
+illustrates the required Vector configuration:
+
+``` toml
+[enrichment_tables.iot_status]
+type = "file"
+
+[enrichment_tables.iot_status.file]
+path = "/etc/vector/iot_status.csv"
+encoding = { type = "csv" }
+
+[enrichment_tables.iot_status.schema]
+status_code = "integer"
+status_message = "string"
+```
+
+After this configuration, we can now translate the output from IoT devices to
+human-readable messages that provide further context in our `iot_status.csv`.
+To do so, we can make use of the [`get_enrichment_table_record`][get_enrichment_table_record] function:
+
+``` toml
+[transforms.enrich_iot_status]
+type = "remap"
+inputs = ["datadog_agents"]
+source = '''
+. = parse_json!(.status_message)
+
+status_code = del(.status_code)
+
+row = get_enrichment_table_record!("iot_status", {"status_code" : status_code})
```

Comment:
I agree with @StephenWakely, for completeness can we show handling the error?

Reply:
Sg. Gonna go with assigning the original value of the status code